addons/mail/static/src/models/messaging_notification_handler/messaging_notification_handler.js

```diff
@@ -455,12 +455,25 @@ function factory(dependencies) {
                 channel.members.includes(this.env.messaging.currentPartner)
             );
 
-            channel = this.env.models['mail.thread'].insert(Object.assign(
-                { isPinned: true },
-                convertedData,
-            ));
+            if (channel) {
+                if (!channel.isPinned && convertedData.serverFoldState === 'closed') {
```

Comment:
What's the spec here, are we supposed to allow unpinned channel from being visible in chat window?

Feels like it shouldn't be possible to have a chat window with a channel that is not visible in the messaging menu (= not pinned).

Reply:
In general we shouldn't allow it, this is a protective measure against bus notifications arriving at a bad time (or from another tab)